### PR TITLE
oui docs - headers page changes

### LINKED ISSUE #110: Header

### PR DESCRIPTION
**Pull request by KrooshalUX**
Signed-off-by: Kroosh Crusius <kroosh@amazon.com>

### Description
Edits to the Header documentation within OUI. These changes include removing references to products no longer relevant to the OpenSearch project. 
 
### Issues Resolved
#110
 
### Check List
- [ ] New functionality includes testing.
- [ ] New functionality has been documented.
- [ ] All tests pass
  - [ ] `yarn lint`
  - [ ] `yarn test-unit`
- [ ] Commits are signed per the DCO using --signoff

By submitting this pull request, I confirm that my contribution is made under the terms of the Apache 2.0 license.

For more information on following Developer Certificate of Origin and signing off your commits, please check [here](https://github.com/opensearch-project/oui/blob/main/CONTRIBUTING.md#developer-certificate-of-origin).

### CODE REVIEW

File: src-docs/src/views/header/header.js [hunk 1]
```diff
@@ -41,7 +37,7 @@ import { htmlIdGenerator } from '../../../../src/services';
 export default () => {
   const renderLogo = () => (
     <OuiHeaderLogo
-      iconType="logoElastic"
+      iconType="logoOUI"
```
Comment (BSFishy) [inline src-docs/src/views/header/header.js:40]: Should these be `logoOpensearch` instead?

As per https://github.com/opensearch-project/oui/issues/111#issuecomment-1397891063

Comment (KrooshalUX) [reply src-docs/src/views/header/header.js:40]: I think so - originally I think there was a miscommunication about the replacement of Elastic to OUI and got swept-up in find>replace instances. I do believe we should be making this logoOpensearch instead, as the logoOUI "namespace" should be used for the actual OUI logo.

Comment (ananzh) [reply src-docs/src/views/header/header.js:40]: @KrooshalUX I think this one is very close to merge. Could you fix this comment?

File: src-docs/src/views/header/header_elastic_pattern.js [hunk 2]
```diff
@@ -283,8 +283,8 @@ export default ({ theme }) => {
         sections={[
           {
             items: [
-              <OuiHeaderLogo iconType="logoElastic" href="">
-                Elastic
+              <OuiHeaderLogo iconType="logoOUI" href="">
```
Comment (BSFishy) [inline src-docs/src/views/header/header_elastic_pattern.js:286]: Still needs to change to `logoOpenSearch`

File: src-docs/src/views/header/header_links.js [hunk 3]
```diff
@@ -23,7 +23,7 @@ export default () => {
   return (
     <OuiHeader>
       <OuiHeaderSectionItem border="right">
-        <OuiHeaderLogo>Elastic</OuiHeaderLogo>
+        <OuiHeaderLogo iconType="logoOUI">OpenSearch</OuiHeaderLogo>
```
Comment (BSFishy) [inline src-docs/src/views/header/header_links.js:26]: Still needs to change to `logoOpenSearch`

File: src-docs/src/views/header/header_position.js [hunk 4]
```diff
@@ -23,7 +23,7 @@ export default () => {
 
   const sections = [
     {
-      items: [<OuiHeaderLogo>Elastic</OuiHeaderLogo>],
+      items: [<OuiHeaderLogo iconType="logoOUI">OpenSearch</OuiHeaderLogo>],
```
Comment (BSFishy) [inline src-docs/src/views/header/header_position.js:26]: Still needs to change to `logoOpenSearch`

File: src-docs/src/views/header/header_sections.js [hunk 5]
```diff
@@ -27,19 +27,13 @@ import {
 export default () => {
   const renderLogo = (
     <OuiHeaderLogo
-      iconType="logoElastic"
+      iconType="logoOUI"
```
Comment (BSFishy) [inline src-docs/src/views/header/header_sections.js:30]: Still needs to change to `logoOpenSearch`

File: src-docs/src/views/header/header_stacked.js [hunk 6]
```diff
@@ -53,14 +53,14 @@ export default () => {
         sections={[
           {
             items: [
-              <OuiHeaderLogo iconType="logoElastic">Elastic</OuiHeaderLogo>,
+              <OuiHeaderLogo iconType="logoOUI">OpenSearch</OuiHeaderLogo>,
```
Comment (BSFishy) [inline src-docs/src/views/header/header_stacked.js:56]: Still needs to change to `logoOpenSearch`

**Review by BSFishy — CHANGES_REQUESTED** [review]
Still have some logos to update

**Comment by KrooshalUX** [comment]
@BSFishy my mistake - somehow only committed one file instead of all of them. This has now been updated and is now available for your review

**Review by BSFishy — CHANGES_REQUESTED** [review]
Sorry to keep passing this back, but linter issues should be fixed before merging

**Review by BSFishy — APPROVED** [review]

**Review by ashwin-pc — APPROVED** [review]

**Pull request merged**